### ISSUE TITLE
For the Gigasecond exercise, introduce the concept of "setup" and "teardown" for the test framework

### ISSUE THREAD

**Issue by Stargator** [exercism/dart#31]
For the Dart exercises, we want to gradually introduce concepts related to the Dart language, test framework, and the built-in tools.

A lot of this is already done, by exposing users to problems written in Dart and asking for a solution. By working to understand the syntax of the Dart language, users will gradually learn more about Dart's features.

But if we keep with the simple setup for test suites from the first exercise, we risk not exposing other capabilities within the test framework. Like groups and the setup/teardown pattern.

So this issue will be just one step in that goal.

Judging by how the test suite for Gigasecond is laid out, I would recommend adding "setup" function to the test suite with comments explaining it's purpose and use. Also, add teardown as well.

There should be comments above the setup and teardown describing it's purpose in the simplest form.

For information about the test framework, [check out the link](https://pub.dartlang.org/packages/test).

**Comment by Stargator** [comment]
@kabiir or @fwip

Once I get your first exercise PR merged in. Then this is the last issue left before we can judge if the track is ready for primetime. Let me know if either of you are interested or have questions.

**Comment by raphalvessa** [comment]
I'm happy to contribute to this issue, I just don't see how the `setUp()` and `tearDown()` would help in this case. I don't see any duplicated code, or code that needs to be ran before and/or after each group test.

**Comment by Stargator** [comment]
@raphalvessa By all means :joy_cat: take this on.

Well, the test suite would have to be reorganized. My thinking was to make some, if not all, the variables `birthDate`, `expectedDate`, `gigasecond` global. Then the setup/teardown could be used to set and re-set. I think the only one setup to handle would be `birthdate`, it would just have to be changed to the same value.

Then again, Gigasecond doesn't **have** to be be the exercise that is reworked for setup/teardown. If there's an exercise you think fits the needs better, go for it!

**Comment by raphalvessa** [comment]
@Stargator Oh right, I thought you were referring exactly to the Leap exercise (though I just noticed that the title mentions the Gigasecond exercise). In that case, I'll look at all the current tests and see how I can improve them. :smiley:

**Comment by Stargator** [comment]
@raphalvessa Not all the tests, just one test suite needs to introduce the setup/teardown

**Comment by raphalvessa** [comment]
@Stargator Oh right, I'll find a good candidate and add it to that one only. Meanwhile, I found ways of optimizing tests that had repeated code (and not because a `setUp()` was missing). Would you like me to include those changes, or should I open an issue?

**Comment by Stargator** [comment]
Open an issue or PR

**Comment by Stargator** [comment]
Related PR was merged in
